internal/pkg/config/x509_test.go

```diff
@@ -64,3 +64,31 @@ func TestPkiCAOutDir(t *testing.T) {
 	assert.True(strings.Contains(pkiDir,
 		filepath.Join(filepath.Base(x509Config.WorkingDir), x509Config.PKISetupDir, x509Config.RootCA.CAName)))
 }
+
+func TestGetCAPemFileName(t *testing.T) {
+	// ignore the file reading error since it will be caught by other test already
+	x509Config, _ := NewX509Config(testJSONFileName)
+	pemFileName := x509Config.GetCAPemFileName()
+	assert.Equal(t, x509Config.RootCA.CAName+"."+certFileExt, pemFileName)
```

Comment:
this isn't a great test since it's duplicating the exact same logic of the function it's testing so if there's some problem with this logic we won't catch it since it's doing the exact same thing that the function is doing to compare the results.
A better test would be:

```go
x509Config, _ := NewX509Config(testJSONFileName)
// code this in the test so if the file changes or the function changes we will catch it in the tests
expected := "the-ca-name.cert"
assert.Equal(t,x509Config.GetCAPemFileName(),expected)
```

Reply:
the second argument of `assert.Equal()` is actually the expected. so I am not following you in the example you gave above.

Reply:
I'm saying construct the expected string without using the X509Config struct

Reply:
ok, i changed all that to hard-coded in test.